ydb/docs/en/core/yql/reference/yql-core/udf/list/datetime.md

```diff
@@ -460,6 +460,23 @@ select
 	DateTime::Format("%Y-%m-%d %Z")(AddTimezone($x, "Europe/Moscow")), -- 2019-10-22 Europe/Moscow
 ```
 
+It's worth mentioning, that several TzDatetime or TzTimestamp values with the positive timezone offset, cannot be casted to TzDate. Consider the example below:
+
+```yql
+select cast(TzDatetime("1970-01-01T23:59:59,Europe/Moscow") as TzDate)
+/* Fatal: Timestamp 1970-01-01T23:59:59.000000,Europe/Moscow cannot be casted to TzDate */
+```
+
+There is no valid value, starting from the Unix epoch, to represent the 01/01/1970 midnight for Europe/Moscow timezone. As a result, such cast is impossible and fails at runtime.
+
+At the same time, values with the negative timezone offset are converted the right way:
```

Comment:
It sounds like this issue doesn't exist for negative timezone offsets, but in reality, it probably still exists around the maximum possible date?

Reply:
First of all, there is no issue with negative timezone offsets.

There is also no way to face the issue with the positive timezone offsets. The datetime value, representing `UINT32_MAX` value, is `2106-02-07T11:28:15Z`. The maximum year value available in YQL is 2106 (non-inclusive), that means the maximum datetime value is `2105-12-31T23:59:59` for every timezone available.

However, 2106-01-01 might be returned for the aforementioned timezones. Consider the following:
```
SELECT CAST(AddTimezone(Datetime("2105-12-31T23:59:59Z"), "Europe/Moscow") as TzDate);
/* 2106-01-01,Europe/Moscow */
```

This value can be represented by UINT32 type, so no issue occurs.